### PR TITLE
[Test] Validate custom-xlator framework

### PR DESCRIPTION
**Pull request by leelavg**
Steps:
- Perform minimal IO on the mount
- Enable custom xlator and verify xlator position in the volfile
- After performing any operation on the custom xlator set options
using 'storage.reserve' to validate other xlators aren't effected
- Add brick to the volume and verify the xlator position in volfile in
the new brick
- Replace brick and verify the xlator position in new brick volfile
- Verify debug xlator is reflected correctly in the volfile when set
- Validate unexisting xlator position should fail
- Reset the volume and verify all the options set above are reset

Change-Id: I5e9dad797fe187b3b9bcdd7374635779c97c5f25
Signed-off-by: Leela Venkaiah G <lgangava@redhat.com>

### CODE REVIEW

**Comment by leelavg** [comment]
please review @itisravi @srijan-sivakumar

**Comment by leelavg** [comment]
Local testing:

```
# pyc tests/functional/glusterd/test_custom_xlator.py                                                    

           Sorting imports by 'isort'
/root/toolbox/venv/gpy3/lib/python3.8/site-packages/setuptools/distutils_patch.py:25: UserWarning: Distutils was imported before Setuptools. This usage is discouraged and may exhibit undesirable behaviors or errors. Please use Setuptools' objects directly or at least import Setuptools first.
  warnings.warn(

           Running through 'yapf'

           Remove unused vars/imports by 'autoflake'

           Checking against 'flake8'

           Errors from 'pylint'

--------------------------------------------------------------------
Your code has been rated at 10.00/10 (previous run: 10.00/10, +0.00)

# On Centos8/RHEL8

tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-replicated_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-arbiter_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-dispersed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_arbiter_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_dispersed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_replicated_glusterfs::test_custom_xlator_ops PASSED

============================= 7 passed, 3849 warnings in 1284.42s (0:21:24) =============================


# On Centos7/RHEL7

tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-dispersed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-replicated_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_distributed-arbiter_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_replicated_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_dispersed_glusterfs::test_custom_xlator_ops PASSED
tests/functional/glusterd/test_custom_xlator.py::TestCustomXlator_cplex_arbiter_glusterfs::test_custom_xlator_ops PASSED

===================================== 7 passed in 1240.35 seconds ======================================
```

**Review by pranavprakash20 — APPROVED** [review]

**Review by balakreddy — APPROVED** [review]

File: tests/functional/glusterd/test_custom_xlator.py [hunk 1]
```diff
@@ -0,0 +1,327 @@
...
+            ret, _, _ = out[host]
+            if ret:
+                raise ExecutionError('Unable to create user xlator directory '
+                                     'or failed to copy custom xlator')
+
+    def setUp(self):
+        self.get_super_method(self, 'setUp')()
+
+        # A single mount is enough for all the tests
+        self.mounts = self.mounts[0:1]
```
Comment (kshithijiyer) [inline tests/functional/glusterd/test_custom_xlator.py:79]: Can't we change it to `self.mounts[0]` as the current code will only have the first element?
```
>>> a = range(1,10)
>>> a
[1, 2, 3, 4, 5, 6, 7, 8, 9]
>>> a[0:1]
[1]
>>> a[0]
1
```

**Review by kshithijiyer — COMMENTED** [review]
Have a minor comment

File: tests/functional/glusterd/test_custom_xlator.py (hunk 1 above)
Comment (leelavg) [reply tests/functional/glusterd/test_custom_xlator.py:79]: It's intentional and the difference can be seen from the example itself.

I actually want a one element *list* which let's me use as it is to `mounts` arg in `setup_volume_and_mount_volume` method

Comment (kshithijiyer) [reply tests/functional/glusterd/test_custom_xlator.py:79]: I would prefer something like `[self.mounts[0]]`, but it's more of a personal preference not really perticular about it.

**Review by kshithijiyer — APPROVED** [review]
LGTM

**Review by balakreddy — APPROVED** [review]

**Review by pranavprakash20 — APPROVED** [review]

**Review by aloganat — APPROVED** [review]

**Pull request merged**